newton/_src/utils/sim_usd.py

```diff
@@ -0,0 +1,310 @@
+            "newton:xpbd:rigid_contact_con_weighting": "rigid_contact_con_weighting",
+            "newton:xpbd:angular_damping": "angular_damping",
+            "newton:xpbd:enable_restitution": "enable_restitution",
+        },
+        IntegratorType.MJWARP: {
+            "newton:mjwarp:use_mujoco_cpu": "use_mujoco_cpu",
+            "newton:mjwarp:solver": "solver",
+            "newton:mjwarp:integrator": "integrator",
+            "newton:mjwarp:iterations": "iterations",
+            "newton:mjwarp:ls_iterations": "ls_iterations",
+            "newton:mjwarp:save_to_mjcf": "save_to_mjcf",
+            "newton:mjwarp:contact_stiffness_time_const": "contact_stiffness_time_const",
+        },
+    }
+    MODEL_ATTRIBUTES_KEYS = MODEL_ATTRIBUTES.keys()
+    SOLVER_ATTRIBUTES_KEYS = SOLVER_ATTRIBUTES.keys()
+
+    def __init__(self, input_path, output_path, integrator: Optional[IntegratorType] = None):
+        def create_stage_from_path(input_path) -> Usd.Stage:
+            stage = Usd.Stage.Open(input_path, Usd.Stage.LoadAll)
+            flattened = stage.Flatten()
+            out_stage = Usd.Stage.Open(flattened.identifier)
+            return out_stage
+
+        self.sim_time = 0.0
+        self.profiler = {}
+
+        self.in_stage = create_stage_from_path(input_path)
+
+        builder = newton.ModelBuilder()
+        builder.up_axis = newton.Axis.Z
+        results = parse_usd(
+            self.in_stage,
+            builder,
+            invert_rotations=True,
+            collapse_fixed_joints=True,
+        )
+
+        scene_attributes = results["scene_attributes"]
+        self._apply_solver_attributes(scene_attributes)
+
+        if integrator:
+            self.integrator_type = integrator
+
+        if self.integrator_type == IntegratorType.VBD:
+            builder.color()
+        self.model = builder.finalize()
+        self.builder_results = results
+        orig_orientation_1 = self.model.body_q.numpy().flatten()[3:]
+        print(f"__init__ orig_orientation_1 = {orig_orientation_1}")
+
+        self.path_body_map = self.builder_results["path_body_map"]
+        collapse_results = self.builder_results["collapse_results"]
+        self.path_body_relative_transform = self.builder_results["path_body_relative_transform"]
+        if collapse_results:
+            self.body_remap = collapse_results["body_remap"]
+            self.body_merged_parent = collapse_results["body_merged_parent"]
+            self.body_merged_transform = collapse_results["body_merged_transform"]
+        else:
+            self.body_remap = None
+            self.body_merged_parent = None
+            self.body_merged_transform = None
+
+        self._apply_model_attributes(scene_attributes)
+        self._setup_integrator(scene_attributes)
+
+        self.state_0 = self.model.state()
+        self.state_1 = self.model.state()
+
+        # NB: body_q will be modified, so initial state will be slightly altered
+        if self.model.joint_count:
```

Comment:
You can remove the if condition, just eval_fk always. It will not do anything if there are no joints.